proposals/csharp-11.0/low-level-struct-improvements.md

```diff
@@ -869,6 +869,149 @@ ref struct S<T>
 }
 ```
 
+### Annotations
+Lifetime models are most naturally expressed via a more explicit annotation model. While the syntax of C# implicitly adds lifetimes to values, there is an underlying model that describes the fundamental rules here. It's often easier to discuss the implication of changes to the design in terms of these rules so they are included here for discussion sake.
+
+Note that this is not meant to be a 100% complete documentation. Documenting every single behavior isn't a goal here. Instead it's meant to establish a general understanding and common verbiage by which the model, and potential changes to it, can be discussed.
+
+The below examples make use of named lifetimes. The syntax `$a` refers to a lifetime named `a`. It is a lifetime that has no meaning by itself but can be given a relationship to other lifetimes via the `where lifetime` syntax. There are a few predefined lifetimes for convenience and brevity below:
+
+- `$heap`: this is the lifetime of any value that exists on the heap
+- `$local`: this is the lifetime of any value that exists on the method stack. It's effectively a name place holder for *current method*
+- `$ro`: name place holder for the *return only* safe to escape scope
+- `$cm`: name place holder for the *calling method* safe to escape scope
+- `$this`: this is the lifetime of the implicit `this` value in an instance method or field declaration.
+
+There are a few predefined relationships between lifetimes:
+
+- `lifetime $heap >= $a` for all lifetimes `$a`
+- `lifetime $cm >= $ro` 
+- `lifetime $x >= $local` for all predefined lifetimes. User defined lifetimes have no relationship to local unless explictly defined.
+- `lifetime $this >= $ro`
+
+Note: while both `$this` and `$cm` are `>= $ro` there is no relationship between the two of them.
+
+The C# method syntax maps to the model in the following ways: 
+
+- `ref` parameters have a ref lifetime of `$ro`
+- parameters of type `ref struct` have a value lifetime of `$cm`
+- ref returns have a ref lifetime of `$ro`
+- returns of type `ref struct` have a value lifetime of `$ro`
+- `scoped` on a parameter or `ref` changes the lifetime to be `$local`
+- `ref this` has lifetime `$local` by default but it becomes `$ro` when the method is annotated with `[UnscopedRef]`
+
+The basic rules for the lifetime are defined as:
+
+- All lifetimes are expressed syntactically as generic arguments, coming before type arguments. This is true for predefined lifetimes except `$heap` and `$local`. 
+- All variables have their lifetime defined at declaration by annotating the type. For example `$a ref $b Span<int> span` defines a local `span` whose value lifetime is `$b` and whose ref lifetime is `$a`. 
+- All values that are not typed to `ref struct` implicitly have a lifetime of `$heap`. That is there is no need to write `$heap int` everywhere, one can simply write `$int`. 
+- The `ref` of a non-ref local has ref lifetime `$local`
```

Comment:
That's true for variables that are not explicitly `ref` but not true otherwise. Happy to hear simpler ways to state this. Kept this statement specific to locals to avoid getting into too many corners here.